applications/RomApplication/python_scripts/hrom_training_utility.py

```diff
@@ -156,28 +159,12 @@ def __CalculateResidualBasis(self):
         return u
 
     def __AppendHRomWeightsToRomParameters(self):
-        n_elements = self.solver.GetComputingModelPart().NumberOfElements()
-        w = np.squeeze(self.hyper_reduction_element_selector.w)
-        z = self.hyper_reduction_element_selector.z
+        number_of_elements = self.solver.GetComputingModelPart().NumberOfElements()
+        weights = np.squeeze(self.hyper_reduction_element_selector.w)
+        indexes = self.hyper_reduction_element_selector.z
 
-        # Create dictionary with HROM weights
-        hrom_weights = {}
-        hrom_weights["Elements"] = {}
-        hrom_weights["Conditions"] = {}
-
-        if type(z)==np.int64 or type(z)==np.int32:
-            # Only one element found !
-            if z <= n_elements-1:
-                hrom_weights["Elements"][int(z)] = float(w)
-            else:
-                hrom_weights["Conditions"][int(z)-n_elements] = float(w)
-        else:
-            # Many elements found
-            for j in range (0,len(z)):
-                if z[j] <=  n_elements -1:
-                    hrom_weights["Elements"][int(z[j])] = float(w[j])
-                else:
-                    hrom_weights["Conditions"][int(z[j])-n_elements] = float(w[j])
+        # Create dictionary with HROM weights (Only used for the expansion of the selected Conditions to include their parent Elements)
+        hrom_weights = self.__CreateDictionaryWithRomElementsAndWeights(weights,indexes,number_of_elements)
```

Comment:
This function is called regardless of the format type ("json" or "numpy"). Inside this function, it tries to read numpy files that may not exist if the user is training with "json".

Reply:
Hyper-reduction training does not work for "json".

Reply:
Now it should work